tools/ci-operator-prowgen/main_test.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"strings"
+	"testing"
+
+	ciop "github.com/openshift/ci-operator/pkg/api"
+)
+
+func checkCondition(t *testing.T, forHumans string, condition bool) {
+	if !condition {
+		t.Errorf("'%s' does not hold", forHumans)
+	}
+}
+
+func checkString(t *testing.T, member, value, expected string) {
+	if value != expected {
+		t.Errorf(
+			"%s not set correctly: expected '%s', got '%s'",
+			member,
+			expected,
+			value,
+		)
+	}
+}
+
+func TestGeneratePodSpec(t *testing.T) {
+	podSpec := generatePodSpec("organization", "repo", "branch", "target")
```

Comment:
Generally you should be able to use `reflect.DeepEqual` for comparison of structs. However, there are semantic meanings to fields in `PodSpec` that exist on top of the normal equality rules for e.g. lists or maps, so we need to use the k8s equality library. We can also use the k8s diff library to get nice diffs on these when they are wrong. See this example: https://github.com/kubernetes/test-infra/blob/master/prow/pod-utils/decorate/podspec_test.go#L711-L713

Reply:
OK